compositor_pipeline/src/pipeline/pipeline_input.rs

```diff
@@ -17,50 +15,52 @@ pub struct PipelineInput {
     pub(super) video_eos_received: Option<bool>,
 }
 
-pub(super) fn register_pipeline_input<NewInputResult>(
-    pipeline: &Arc<Mutex<Pipeline>>,
-    input_id: InputId,
-    input_options: &dyn InputOptionsExt<NewInputResult>,
-    queue_options: QueueInputOptions,
-) -> Result<NewInputResult, RegisterInputError> {
-    let pipeline_ctx = pipeline.lock().unwrap().ctx.clone();
```

Comment:
maybe add comment here that we intentionally do not hold lock here so we don't block the processing. I think I saw change like that in PR a few times already